lib/v2/reuters/common.js

```diff
@@ -9,31 +9,54 @@ module.exports = async (ctx) => {
     const topic = ctx.params.topic ?? (category === 'authors' ? 'reuters' : '');
     const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 20;
 
-    const rootUrl = 'https://www.reuters.com';
-    const currentUrl = topic ? `${rootUrl}/${category}/${topic}/` : `${rootUrl}/${category}/`;
-    const response = await got(currentUrl);
-    const $ = cheerio.load(response.data);
-
-    let items = $('.media-story-card__body__3tRWy a.media-story-card__heading__eqhp9, a.svelte-pxbp38, a.svelte-11dknnx, a.svelte-e21rsn')
-        .slice(0, limit)
-        .toArray()
-        .map((item) => {
-            item = $(item);
-            item.find('span.visually-hidden__hidden__2qXMW').remove();
+    const MUST_FETCH_BY_TOPICS = ['authors'];
+    const section_id = `/${category}/${topic ? `${topic}/` : ''}`;
+    const { title, description, rootUrl, response } = await (async () => {
+        if (!MUST_FETCH_BY_TOPICS.includes(category)) {
+            const rootUrl = 'https://www.reuters.com/pf/api/v3/content/fetch/articles-by-section-alias-or-id-v1';
+            const response = await got(
+                rootUrl, {searchParams:`query={${encodeURI(
+                    JSON.stringify({
+                        offset: 0,
+                        size: limit,
+                        section_id,
+                        website: 'reuters',
+                    }).slice(1, -1)
+                )}}`}
```

Comment:
https://github.com/DIYgod/RSSHub/blob/5cef317a43b78405f76eb7f2fa86015c467155c4/lib/v2/eastmoney/search/index.js#L33

Reply:
I tried, and the two bracket will be encoded as well, causing error.

Reply:
https://github.com/DIYgod/RSSHub/commit/e520080ee805d47d25ad32695c2423e3bf5e5636

Reply:
![image](https://github.com/DIYgod/RSSHub/assets/31068367/2e582e10-e3e6-4941-91d5-d84c8794b135)

Reply:
> You **don't** need to do <ins>`key=value`</ins> and <ins>manually encode it</ins>. It will be handled by got.

https://github.com/DIYgod/RSSHub/blob/5cef317a43b78405f76eb7f2fa86015c467155c4/lib/v2/eastmoney/search/index.js#L33

```js
                searchParams: {
                    query: JSON.stringify({
                        offset: 0,
                        size: limit,
                        section_id,
                        website: 'reuters',
                    }),
                }
```

Reply:
Gosh I'm sorry...